pkg/collector/corechecks/cluster/kubernetes_state.go

```diff
@@ -339,65 +328,26 @@ func (k *KSMCheck) metricFilter(m ksmstore.DDMetric) bool {
 	return m.Val == float64(1)
 }
 
-// isMatching returns whether a targeted metric for label joins is
-// matching another metric labels based on the labels to match config
-func isMatching(config *JoinsConfig, destlabels, srcLabels map[string]string) bool {
-	for _, l := range config.LabelsToMatch {
-		firstVal, found := destlabels[l]
-		if !found {
-			return false
-		}
-		secondVal, found := srcLabels[l]
-		if !found {
-			return false
-		}
-		if firstVal != secondVal {
-			return false
-		}
-	}
-	return true
-}
-
 // buildTag applies the LabelsMapper config and returns the tag in a key:value string format
 // The second return value is the hostname of the metric if a 'node' or 'host' tag is found, empty string otherwise
 func (k *KSMCheck) buildTag(key, value string) (tag, hostname string) {
 	if newKey, found := k.instance.LabelsMapper[key]; found {
 		key = newKey
 	}
-	tag = fmt.Sprintf("%s:%s", key, value)
+
+	var sb strings.Builder
+	sb.Grow(len(key) + 1 + len(value))
+	sb.WriteString(key)
+	sb.WriteRune(':')
```

Comment:
I wonder if replacing `strings.Builder` with just a byte slice makes any difference here? It might be ridiculous but it at least would skip the utf8 encoding check here - https://golang.org/src/strings/builder.go?s=386:479#L106

Reply:
`strings.Builder` is already handling a byte slice behind the hood.
So, removing the `strings.Builder` abstraction may not provide a visible effect.
But I can definitively get rid of all the UTF-8 logic by replacing the `WriteRune` by `WriteByte` !